app/models/talk.rb

```diff
@@ -286,4 +290,20 @@ def suggestion_summary
       at #{event.name}
     HEREDOC
   end
+
+  def set_kind
+    self.kind =
+      case title
+      when /.*(keynote:|opening\ keynote|closing\ keynote).*/i
+        :keynote
+      when /.*workshop:.*/i
+        :workshop
+      when /.*panel:.*/i
+        :panel
+      when /.*lightning\ talk: .*/i
+        :lightning_talk
+      else
+        :standard
+      end
+  end
```

Comment:
I wonder if we should keep a way to force the kind?

```ruby
Talk.create(title: "some title of a group talk", kind: "panel)
```

not sure if there is (yet) a real use case ?

Reply:
@adrienpoly, we thought about it, and the only use case we can think of is if you add a kind in the YML files or create a talk directly from the console.

What do you about adding this to the callback:

```ruby
before_action :set_kind, if: -> { !kind_changed? }
```

Following @marcoroth’s suggestion to add title_changed?, we can include both conditions:

```ruby
before_action :set_kind, if: -> { title_changed? && !kind_changed? }
```

Reply:
I'm running into an issue with using `title_changed` as a condition in the callback. When we run seeds, we need to update existing talks, but it never triggers `set_kind` because the title hasn't changed.

I think it’s better to keep just the !kind_changed? condition so we can skip the callback if a kind is provided.

Thoughts? @marcoroth @adrienpoly

Reply:
good for me